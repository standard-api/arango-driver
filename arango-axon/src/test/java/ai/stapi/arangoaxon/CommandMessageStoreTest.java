package ai.stapi.arangoaxon;

import ai.stapi.graphsystem.messaging.command.DynamicCommand;
import ai.stapi.identity.UniqueIdentifier;
import ai.stapi.objectRenderer.infrastructure.objectToJsonStringRenderer.ObjectToJSonStringOptions.RenderFeature;
import ai.stapi.test.integration.IntegrationTestCase;
import java.util.HashMap;
import java.util.Map;
import org.axonframework.commandhandling.GenericCommandMessage;
import org.axonframework.modelling.command.TargetAggregateIdentifier;
import org.junit.jupiter.api.BeforeEach;
import org.junit.jupiter.api.Test;
import org.springframework.beans.factory.annotation.Autowired;

class CommandMessageStoreTest extends IntegrationTestCase {

  @Autowired
  private ArangoCommandMessageStore commandMessageStore;


  @BeforeEach
  void clean() {
    this.commandMessageStore.wipeAll();
  }

  @Test
  void itCanStoreAndLoadPersistedCommandMessages() {
    this.commandMessageStore.storeCommand(
        new GenericCommandMessage<>(
            new ExampleCommand("Id", "someValue")
        )
    );

    this.commandMessageStore.storeCommand(
        new GenericCommandMessage<>(
            new GenericCommandMessage<>(
                new DynamicCommand(
                    new UniqueIdentifier("ExampleDynamicCommmandId"),
                    "ExampledDynamicCommandType",
                    new HashMap<>(Map.of(
                        "exampleData", "exampleValue"
                    ))
                )
            ),
            "ExampledDynamicCommandType"
        )
    );
    this.thenObjectApproved(
        this.commandMessageStore.getAll(),
        RenderFeature.HIDE_KEY_HASHCODE,
        RenderFeature.SORT_FIELDS
    );
  }
  
  private static class ExampleCommand {
    
    @TargetAggregateIdentifier
    private final String identifier;
    private final String someField;

    public ExampleCommand(String identifier, String someField) {
      this.identifier = identifier;
      this.someField = someField;
    }

    public String getIdentifier() {
      return identifier;
    }

    public String getSomeField() {
      return someField;
    }
  }
}