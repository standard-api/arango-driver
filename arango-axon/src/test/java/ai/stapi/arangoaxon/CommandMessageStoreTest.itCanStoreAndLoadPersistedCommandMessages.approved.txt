[ {
  "targetAggregateIdentifier" : "Id",
  "commandName" : "ai.stapi.arangoaxon.CommandMessageStoreTest$ExampleCommand",
  "commandMetaData" : { },
  "commandPayload" : {
    "identifier" : "Id",
    "someField" : "someValue"
  }
}, {
  "targetAggregateIdentifier" : "ExampleDynamicCommmandId",
  "commandName" : "ExampledDynamicCommandType",
  "commandMetaData" : { },
  "commandPayload" : {
    "id" : "ExampleDynamicCommmandId",
    "exampleData" : "exampleValue"
  }
} ]